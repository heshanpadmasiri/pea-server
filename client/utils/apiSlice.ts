import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
import config from '../config.json';

export interface Metadata {
    name: string,
    id: string,
    ty: string,
    tags: string[],
}

export type Condition = {
    type: string,
    tags: string[]
}

export type QueryResult = {
    isLoading: boolean,
    isError: boolean,
    isSuccess: boolean,
    error: any,
    data: Metadata[]
}

export function fileContentUrl(file: Metadata): string {
    return config.SERVER_URL + '/content/' + file.id;
}

export const apiSlice = createApi({
    reducerPath: 'api',
    baseQuery: fetchBaseQuery({ baseUrl: config.SERVER_URL }),
    endpoints: (builder) => ({
        getTags: builder.query<string[], void>({
            query: () => '/tags'
        }),
        getFiles: builder.query<Metadata[], void>({
            query: () => '/files'
        }),
        getFilesByType: builder.query<Metadata[], string>({
            query: (type) => `/files/${type}`
        }),
        getFilesByCondition: builder.query<Metadata[], Condition>({
            query: ({ type, tags }) => ({
                url: '/query',
                method: 'POST',
                body: {
                    data: { tags, ty: type }
                }
            })
        })
    })
});

export const { useGetTagsQuery, useGetFilesQuery, useGetFilesByTypeQuery, useGetFilesByConditionQuery } = apiSlice;
