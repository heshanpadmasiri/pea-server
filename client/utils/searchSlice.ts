import { createSlice } from '@reduxjs/toolkit';

export const searchSlice = createSlice({
    name: 'search',
    initialState: {
        query: <undefined | string>undefined,
    },
    reducers: {
        clearQuery: (state) => {
            state.query = undefined;
        },
        setQuery: (state, action) => {
            const newQuery = action.payload as string;
            state.query = newQuery;
        }
    }
})

export const { clearQuery, setQuery } = searchSlice.actions;
export default searchSlice.reducer;
