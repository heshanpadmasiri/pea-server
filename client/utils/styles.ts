import { StyleSheet } from 'react-native';
const styles = StyleSheet.create({
    safeArea: {
        flex: 5,
        paddingHorizontal: 10,
    },
    searchBar: {
        flexDirection: 'row',
        flex: 0.25,
        alignItems: 'center',
        justifyContent: 'space-between',
    },
    gallery: {
        flex: 1,
        backgroundColor: 'black'
    },
    grid: {
        flexDirection: 'row',
    },
    grid_image: {
        height: 300,
        width: 300,
        resizeMode:'contain',
    },
    tag_selector: {
        flex: 3,
    },
    switch_container: {
        flex: 1,
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'space-between',
    },
    thumbnailCard: {
        alignItems: 'center',
        backgroundColor: 'white',
        marginBottom: 10,
        height: 220,
    },
    thumbnailHeading: {
        flex: 1
    },
    thumbnail: {
        height: 200,
        width: 200
    },
    container: {
        flex: 1,
        justifyContent: 'center',
        backgroundColor: '#ecf0f1',
    },
    video: {
        alignSelf: 'center',
        width: 320,
        height: 200,
    },
    buttons: {
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
    },
});


export default styles;
