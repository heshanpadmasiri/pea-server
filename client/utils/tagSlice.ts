import { createSlice } from "@reduxjs/toolkit";

export const tagSlice = createSlice({
    name: "tags",
    initialState: {
        selectedTags: <string[]>[],
    },
    reducers: {
        selectTag: (state, action) => {
            if (state.selectedTags.includes(action.payload)) {
                return;
            }
            state.selectedTags.push(action.payload);
        },
        unselectTag: (state, action) => {
            state.selectedTags = state.selectedTags.filter((tag) => tag !== action.payload);
        },
    }
});

export const { selectTag, unselectTag } = tagSlice.actions;

export default tagSlice.reducer;
