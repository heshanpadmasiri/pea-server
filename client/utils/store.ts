import { configureStore } from '@reduxjs/toolkit';
import tagReducer from './tagSlice';
import slideShowReducer from './slideShowSlice';
import { apiSlice } from './apiSlice';

const store = configureStore({
    reducer: {
        tags: tagReducer,
        slideShow: slideShowReducer,
        [apiSlice.reducerPath]: apiSlice.reducer
    },
    middleware: getDefaultMiddleware => getDefaultMiddleware(
        { serializableCheck: false }
    ).concat(apiSlice.middleware)
});

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch

export default store;
