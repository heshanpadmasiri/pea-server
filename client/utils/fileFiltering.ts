import { useSelector } from 'react-redux';
import { QueryResult, useGetFilesQuery } from './apiSlice';
import { RootState } from './store';
import { useGetFilesByConditionQuery, useGetFilesByTypeQuery } from './apiSlice';

export function getFilesByType(type: string): QueryResult {
    const selectedTags = useSelector((state: RootState) => state.tags.selectedTags);
    let queryResult: QueryResult;
    if (selectedTags.length > 0) {
        queryResult = useGetFilesByConditionQuery({ type, tags: selectedTags }) as QueryResult;
    }
    else {
        queryResult = useGetFilesByTypeQuery(type) as QueryResult;
    }
    return filterBySearchQuery(queryResult);
}

export function getFilesByTypes(types: string[]): QueryResult[] {
    return types.map((type) => getFilesByType(type))
}

export function getAllFiles(): QueryResult {
    const selectedTags = useSelector((state: RootState) => state.tags.selectedTags);
    let queryResult: QueryResult;
    if (selectedTags.length > 0) {
        queryResult = useGetFilesByConditionQuery({ type: '', tags: selectedTags }) as QueryResult;
    }
    else {
        queryResult = useGetFilesQuery() as QueryResult;
    }
    return filterBySearchQuery(queryResult);
}

function filterBySearchQuery(queryResult: QueryResult): QueryResult {
    const query = useSelector((state: RootState) => state.search.query);
    if (query === undefined) {
        return queryResult;
    }
    if (queryResult.isError || queryResult.isLoading) {
        return queryResult;
    }
    const data = queryResult.data.filter((file) => file.name.toLowerCase().includes(query.toLowerCase()));
    return { ...queryResult, data };
}
