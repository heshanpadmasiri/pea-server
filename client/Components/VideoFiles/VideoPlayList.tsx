import { FlatList, Text, View } from 'react-native';
import styles from '../../utils/styles';
import { fileContentUrl, Metadata } from '../../utils/apiSlice';
import React from 'react';
import { Video } from 'expo-av';
import * as ScreenOrientation from 'expo-screen-orientation';
import { getFilesByType } from '../../utils/fileFiltering';

export default function VideoPlayList() {
    let result = getFilesByType('mp4');
    let content;
    if (result.isLoading) {
        content = (
            <Text>Loading...</Text>
        );
    }
    else if (result.isError) {
        console.error(result.error);
        content = (
            <Text>Error!</Text>
        )
    }
    else if (result.isSuccess) {
        const thumbnails = result.data.map((file: Metadata, index: number) => {
            return {
                index,
                id: file.id,
                url: fileContentUrl(file),
                title: file.name,
            }
        });
        content = (
            <FlatList data={thumbnails} renderItem={({ item }) => <ThumbnailCard {...item} />} />
        )
    }
    return (
        <View style={styles.safeArea}>
            {content}
        </View>
    )
}

type ThumbnailCardProps = {
    index: number;
    title: string
    url: string,
};


function ThumbnailCard(props: ThumbnailCardProps) {
    const title = props.title;
    return (
        <View style={styles.thumbnailCard}>
            <Text style={styles.thumbnailHeading}>{title}</Text>
            <VideoPlayer {...props} />
        </View>
    );
}

type VideoPlayerProps = {
    url: string;
}

function VideoPlayer(props: VideoPlayerProps) {
    const { url } = props;
    const source = { uri: url };
    const video = React.useRef<Video>(null);
    const setOrientation = () => {
        ScreenOrientation.lockAsync(ScreenOrientation.OrientationLock.LANDSCAPE);
    }
    return (
        <Video
            ref={video}
            style={styles.video}
            source={source}
            useNativeControls
            isLooping
            onFullscreenUpdate={setOrientation}
        />
    );
}
