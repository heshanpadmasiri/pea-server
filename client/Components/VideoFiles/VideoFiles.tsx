import { SafeAreaView } from 'react-native';
import styles from '../../utils/styles';
import TagSelector from '../TagSelector';
import VideoPlayList from './VideoPlayList';
import React from 'react';

export default function VideoFiles() {
    return (
        <SafeAreaView style={styles.safeArea}>
            <TagSelector/>
            <VideoPlayList/>
        </SafeAreaView>
    )
}
