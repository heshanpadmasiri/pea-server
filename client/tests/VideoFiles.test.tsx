import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { fireEvent, render, screen, waitForElementToBeRemoved } from '@testing-library/react-native'
import config from '../config.json';
import { setupStore } from '../utils/store';
import { Provider } from 'react-redux';
import VideoFiles from '../Components/VideoFiles/VideoFiles';

const BASE_URL = config.SERVER_URL;
const TIMEOUT = 50000;

export const handlers = [
    rest.get(`${BASE_URL}/tags`, (_req, res, ctx) => {
        return res(ctx.json(['tag 1', 'tag 2']), ctx.status(200));
    }),
    rest.get(`${BASE_URL}/files/mp4`, (_req, res, ctx) => {
        return res(ctx.json([
            {
                name: 'file 1.mp4',
                id: '1',
                ty: 'mp4',
                tags: []
            },
            {
                name: 'file 2.mp4',
                id: '2',
                ty: 'mp4',
                tags: ['tag 1']
            },
            {
                name: 'file 3.mp4',
                id: '3',
                ty: 'mp4',
                tags: ['tag 1', 'tag 2']
            },
        ]), ctx.status(200));
    }),
    rest.post(`${BASE_URL}/query`, async (req, res, ctx) => {
        const data = (await req.json())['data'];
        const ty: string = data['ty'] as string;
        const tags: string[] = data['tags'] as string[];
        if (tags.length == 1 && tags[0] == 'tag 1' && ty == 'mp4') {
            return res(ctx.json([
                {
                    name: 'file 2.mp4',
                    id: '2',
                    ty: 'mp4',
                    tags: ['tag 1']
                },
                {
                    name: 'file 3.mp4',
                    id: '3',
                    ty: 'mp4',
                    tags: ['tag 1', 'tag 2']
                },
            ]), ctx.status(200));
        }
        else {
            return res(ctx.json('unexpected request'), ctx.status(500));
        }
    })
]

const server = setupServer(...handlers);
beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close(), TIMEOUT);

describe('VideoFiles', () => {

    test('renders component', async () => {
        const store = setupStore();
        render(
            <Provider store={store}>
                <VideoFiles />
            </Provider>
        );
        await screen.findByTestId('received-tag', {}, { timeout: TIMEOUT });
        await screen.findByTestId('received-files', {}, { timeout: TIMEOUT });
        expect(screen.toJSON()).toMatchSnapshot()
    }, TIMEOUT);

    test('tag selection works', async () => {
        const store = setupStore();
        render(
            <Provider store={store}>
                <VideoFiles />
            </Provider>
        );
        await screen.findByTestId('received-tag', {}, { timeout: TIMEOUT });
        const tag1_selector = await screen.findByTestId('switch-tag 1', {}, { timeout: TIMEOUT });
        fireEvent(tag1_selector, 'onValueChange', true);
        await screen.findByTestId('received-files', {}, { timeout: TIMEOUT });
        const file_1 = await screen.queryByText('file 1.txt');
        if (file_1 != null) {
            await waitForElementToBeRemoved(() => file_1, { timeout: TIMEOUT });
        }
        expect(screen.toJSON()).toMatchSnapshot()
    }, TIMEOUT);
});
