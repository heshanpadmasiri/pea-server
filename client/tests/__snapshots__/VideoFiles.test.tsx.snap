// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoFiles renders component 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 5,
      "paddingHorizontal": 10,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 0.25,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <TextInput
      defaultValue="search"
      onChangeText={[Function]}
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": undefined,
              "fontSize": 24,
            },
            undefined,
            {
              "fontFamily": "ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        ïˆ¥
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": "#ecf0f1",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <RCTScrollView
      style={
        {
          "flex": 3,
        }
      }
      testID="received-tag"
    >
      <View>
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text>
            tag 1
          </Text>
          <RCTSwitch
            accessibilityRole="switch"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTintColor="#81b0ff"
            style={
              [
                {
                  "height": 31,
                  "width": 51,
                },
                {
                  "backgroundColor": "#3e3e3e",
                  "borderRadius": 16,
                },
              ]
            }
            testID="switch-tag 1"
            thumbTintColor="#f4f3f4"
            tintColor="#767577"
            value={false}
          />
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text>
            tag 2
          </Text>
          <RCTSwitch
            accessibilityRole="switch"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTintColor="#81b0ff"
            style={
              [
                {
                  "height": 31,
                  "width": 51,
                },
                {
                  "backgroundColor": "#3e3e3e",
                  "borderRadius": 16,
                },
              ]
            }
            testID="switch-tag 2"
            thumbTintColor="#f4f3f4"
            tintColor="#767577"
            value={false}
          />
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      {
        "flex": 5,
        "paddingHorizontal": 10,
      }
    }
  >
    <RCTScrollView
      data={
        [
          {
            "id": "1",
            "index": 0,
            "title": "file 1.mp4",
            "url": "http://192.168.8.176:8080/content/1",
          },
          {
            "id": "2",
            "index": 1,
            "title": "file 2.mp4",
            "url": "http://192.168.8.176:8080/content/2",
          },
          {
            "id": "3",
            "index": 2,
            "title": "file 3.mp4",
            "url": "http://192.168.8.176:8080/content/3",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      testID="received-files"
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "white",
                "height": 220,
                "marginBottom": 10,
              }
            }
          >
            <Text
              style={
                {
                  "flex": 1,
                }
              }
            >
              file 1.mp4
            </Text>
            <View
              pointerEvents="box-none"
              style={
                {
                  "alignSelf": "center",
                  "height": 200,
                  "overflow": "hidden",
                  "width": 320,
                }
              }
            >
              <ViewManagerAdapter_ExpoVideoView
                onError={[Function]}
                onFullscreenUpdate={[Function]}
                onLoad={[Function]}
                onLoadStart={[Function]}
                onReadyForDisplay={[Function]}
                onStatusUpdate={[Function]}
                proxiedProperties={
                  {
                    "resizeMode": "mock",
                    "source": {
                      "headers": undefined,
                      "overridingExtension": null,
                      "uri": "http://192.168.8.176:8080/content/1",
                    },
                    "status": {
                      "isLooping": true,
                    },
                    "useNativeControls": true,
                  }
                }
                style={
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
                videoStyle={
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              />
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "white",
                "height": 220,
                "marginBottom": 10,
              }
            }
          >
            <Text
              style={
                {
                  "flex": 1,
                }
              }
            >
              file 2.mp4
            </Text>
            <View
              pointerEvents="box-none"
              style={
                {
                  "alignSelf": "center",
                  "height": 200,
                  "overflow": "hidden",
                  "width": 320,
                }
              }
            >
              <ViewManagerAdapter_ExpoVideoView
                onError={[Function]}
                onFullscreenUpdate={[Function]}
                onLoad={[Function]}
                onLoadStart={[Function]}
                onReadyForDisplay={[Function]}
                onStatusUpdate={[Function]}
                proxiedProperties={
                  {
                    "resizeMode": "mock",
                    "source": {
                      "headers": undefined,
                      "overridingExtension": null,
                      "uri": "http://192.168.8.176:8080/content/2",
                    },
                    "status": {
                      "isLooping": true,
                    },
                    "useNativeControls": true,
                  }
                }
                style={
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
                videoStyle={
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              />
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "white",
                "height": 220,
                "marginBottom": 10,
              }
            }
          >
            <Text
              style={
                {
                  "flex": 1,
                }
              }
            >
              file 3.mp4
            </Text>
            <View
              pointerEvents="box-none"
              style={
                {
                  "alignSelf": "center",
                  "height": 200,
                  "overflow": "hidden",
                  "width": 320,
                }
              }
            >
              <ViewManagerAdapter_ExpoVideoView
                onError={[Function]}
                onFullscreenUpdate={[Function]}
                onLoad={[Function]}
                onLoadStart={[Function]}
                onReadyForDisplay={[Function]}
                onStatusUpdate={[Function]}
                proxiedProperties={
                  {
                    "resizeMode": "mock",
                    "source": {
                      "headers": undefined,
                      "overridingExtension": null,
                      "uri": "http://192.168.8.176:8080/content/3",
                    },
                    "status": {
                      "isLooping": true,
                    },
                    "useNativeControls": true,
                  }
                }
                style={
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
                videoStyle={
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;

exports[`VideoFiles tag selection works 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 5,
      "paddingHorizontal": 10,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 0.25,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <TextInput
      defaultValue="search"
      onChangeText={[Function]}
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": undefined,
              "fontSize": 24,
            },
            undefined,
            {
              "fontFamily": "ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        ïˆ¥
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": "#ecf0f1",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <RCTScrollView
      style={
        {
          "flex": 3,
        }
      }
      testID="received-tag"
    >
      <View>
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text>
            tag 1
          </Text>
          <RCTSwitch
            accessibilityRole="switch"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTintColor="#81b0ff"
            style={
              [
                {
                  "height": 31,
                  "width": 51,
                },
                {
                  "backgroundColor": "#3e3e3e",
                  "borderRadius": 16,
                },
              ]
            }
            testID="switch-tag 1"
            thumbTintColor="#f5dd4b"
            tintColor="#767577"
            value={true}
          />
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text>
            tag 2
          </Text>
          <RCTSwitch
            accessibilityRole="switch"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTintColor="#81b0ff"
            style={
              [
                {
                  "height": 31,
                  "width": 51,
                },
                {
                  "backgroundColor": "#3e3e3e",
                  "borderRadius": 16,
                },
              ]
            }
            testID="switch-tag 2"
            thumbTintColor="#f4f3f4"
            tintColor="#767577"
            value={false}
          />
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      {
        "flex": 5,
        "paddingHorizontal": 10,
      }
    }
  >
    <RCTScrollView
      data={
        [
          {
            "id": "2",
            "index": 0,
            "title": "file 2.mp4",
            "url": "http://192.168.8.176:8080/content/2",
          },
          {
            "id": "3",
            "index": 1,
            "title": "file 3.mp4",
            "url": "http://192.168.8.176:8080/content/3",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      testID="received-files"
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "white",
                "height": 220,
                "marginBottom": 10,
              }
            }
          >
            <Text
              style={
                {
                  "flex": 1,
                }
              }
            >
              file 2.mp4
            </Text>
            <View
              pointerEvents="box-none"
              style={
                {
                  "alignSelf": "center",
                  "height": 200,
                  "overflow": "hidden",
                  "width": 320,
                }
              }
            >
              <ViewManagerAdapter_ExpoVideoView
                onError={[Function]}
                onFullscreenUpdate={[Function]}
                onLoad={[Function]}
                onLoadStart={[Function]}
                onReadyForDisplay={[Function]}
                onStatusUpdate={[Function]}
                proxiedProperties={
                  {
                    "resizeMode": "mock",
                    "source": {
                      "headers": undefined,
                      "overridingExtension": null,
                      "uri": "http://192.168.8.176:8080/content/2",
                    },
                    "status": {
                      "isLooping": true,
                    },
                    "useNativeControls": true,
                  }
                }
                style={
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
                videoStyle={
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              />
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "white",
                "height": 220,
                "marginBottom": 10,
              }
            }
          >
            <Text
              style={
                {
                  "flex": 1,
                }
              }
            >
              file 3.mp4
            </Text>
            <View
              pointerEvents="box-none"
              style={
                {
                  "alignSelf": "center",
                  "height": 200,
                  "overflow": "hidden",
                  "width": 320,
                }
              }
            >
              <ViewManagerAdapter_ExpoVideoView
                onError={[Function]}
                onFullscreenUpdate={[Function]}
                onLoad={[Function]}
                onLoadStart={[Function]}
                onReadyForDisplay={[Function]}
                onStatusUpdate={[Function]}
                proxiedProperties={
                  {
                    "resizeMode": "mock",
                    "source": {
                      "headers": undefined,
                      "overridingExtension": null,
                      "uri": "http://192.168.8.176:8080/content/3",
                    },
                    "status": {
                      "isLooping": true,
                    },
                    "useNativeControls": true,
                  }
                }
                style={
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
                videoStyle={
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
